controllers/base_controller.go

```diff
@@ -98,47 +98,103 @@ func (r *BaseReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	spiffeId := r.makeSpiffeId(obj)
-	parentId := r.makeParentId(obj)
-	if spiffeId == "" || parentId == "" {
+	myEntry := r.makeEntryForObject(obj)
+
+	if myEntry == nil {
 		// Object does not need an entry. This might be a change, so we need to delete any hanging entries.
+		reqLogger.V(1).Info("Deleting entries for pod that no longer needs an ID", "count", len(matchedEntries))
 		err := r.deleteAllEntries(ctx, reqLogger, matchedEntries)
 		return ctrl.Result{}, err
 	}
 
-	createEntryIfNotExistsResponse, err := r.SpireClient.CreateEntryIfNotExists(ctx, &common.RegistrationEntry{
-		Selectors: r.getSelectors(req.NamespacedName),
-		ParentId:  r.makeParentId(obj),
-		SpiffeId:  spiffeId,
-	})
+	var myEntryId string
 
-	if err != nil {
-		reqLogger.Error(err, "Failed to create or update spire entry")
-		return ctrl.Result{}, err
-	}
-	if !createEntryIfNotExistsResponse.Preexisting {
-		reqLogger.Info("Created new spire entry", "entry", createEntryIfNotExistsResponse.Entry)
+	if len(matchedEntries) == 0 {
+		createEntryIfNotExistsResponse, err := r.SpireClient.CreateEntryIfNotExists(ctx, myEntry)
+		if err != nil {
+			reqLogger.Error(err, "Failed to create or update spire entry")
+			return ctrl.Result{}, err
+		}
+		if !createEntryIfNotExistsResponse.Preexisting {
+			reqLogger.Info("Created new spire entry", "entry", createEntryIfNotExistsResponse.Entry)
+		} else {
+			reqLogger.V(1).Info("Found existing identical spire entry", "entry", createEntryIfNotExistsResponse.Entry)
+		}
+		myEntryId = createEntryIfNotExistsResponse.Entry.EntryId
+	} else {
+		// One of the existing entries might already be just right for me... if not, we can choose one and update it
```

Comment:
Uneducated reader question: if there were matches, how can there be no matching entry? Would be nice to add a sentence about this, or point somewhere to find out.

Reply:
The "matchedEntries" have selectors which would match this pod, and look like they were created by the registrar (based on the parent.) In normal operation you'd expect only one match, and that'll either perfectly match the current pod configuration, or need some minor update (spiffe ID has changed, or dnsNames need updating.)

You'd find more than one only under weird conditions, perhaps someone is running multiple registrars without leader election, or a crash at an inopportune moment.